### ISSUE TITLE
rpm package broken (permissions)

### ISSUE THREAD

**Issue by philfry** [influxdata/influxdb#23850]
Hi,

the rpm package for influxdb2 is severely broken regarding permissions.
The systemd unit file runs influxdb as `influxdb:influxdb` but this user has no permissions at all for
- the script that's being invoked by systemd
- the influxd binary
- the data directory

See:
```
$ rpm -qp --dump influxdb2-2.5.0.x86_64.rpm
/etc/logrotate.d/influxdb 113 1667321990 546afc8a077d25862a8c0a4c1220223f 0100750 root root 0 0 0 X
/usr/bin/influxd 103913640 1667321955 e99394de7d1efe3d49e51c8206162f57 0100750 root root 0 0 0 X
/usr/lib/influxdb/scripts/influxd-systemd-start.sh 872 1667321990 b4fa432fa8c0932bb07a36789c4c1a28 0100750 root root 0 0 0 X
/usr/lib/influxdb/scripts/influxdb.service 545 1667321990 3907778e881b00b028dea7082a0e95a4 0100750 root root 0 0 0 X
/usr/lib/influxdb/scripts/init.sh 5521 1667321990 4ed16779bfa5259d058e304b20c8db74 0100750 root root 0 0 0 X
/usr/share/influxdb/influxdb2-upgrade.sh 2393 1667321990 0af94caf8f73813cc8c25e7752b1f9e1 0100750 root root 0 0 0 X
/var/lib/influxdb 0 1667322035 00000000000000000000000000000000 040750 root root 0 0 0 X
/var/lib/influxdb/.keep 0 1667321990 d41d8cd98f00b204e9800998ecf8427e 0100750 root root 0 0 0 X
```

- `/usr/bin/influxd` should be `%attr(0755, root, root)`
- `/usr/lib/influxdb/scripts/influxd-systemd-start.sh` should be `%attr(0755, root, root)`
- `/usr/lib/influxdb/scripts/influxdb.service` should be `%attr(0644, root, root)` (this didn't break anything, it's just bad style)
- `/etc/logrotate.d/influxdb` should be `%attr(0644, root, root)` (same here: bad style)
- `/usr/lib/influxdb/scripts/init.sh` should be `%attr(0755, root, root)`
- `/usr/share/influxdb/influxdb2-upgrade.sh` should be `%attr(0755, root, root)`
- `/var/lib/influxdb` should be `%attr(0750, influxdb, influxdb) %dir`
- `/var/lib/influxdb/.keep` can be removed from package

With 2.4.0 the data directory was not packaged at all:
```
$ rpm -qp --dump influxdb2-2.4.0.x86_64.rpm 
/etc/logrotate.d/influxdb 113 1660851879 546afc8a077d25862a8c0a4c1220223f 0100664 root root 0 0 0 X
/usr/bin/influxd 150164784 1660851855 a7e42d3f2031847e83bb753ecc311ea3 0100755 root root 0 0 0 X
/usr/lib/influxdb/scripts/influxd-systemd-start.sh 872 1660851879 b4fa432fa8c0932bb07a36789c4c1a28 0100775 root root 0 0 0 X
/usr/lib/influxdb/scripts/influxdb.service 534 1660851879 ed443c93fe9571350a765becea9cb086 0100664 root root 0 0 0 X
/usr/lib/influxdb/scripts/init.sh 5521 1660851879 4ed16779bfa5259d058e304b20c8db74 0100664 root root 0 0 0 X
/usr/share/influxdb/influxdb2-upgrade.sh 2393 1660851879 0af94caf8f73813cc8c25e7752b1f9e1 0100775 root root 0 0 0 X
```
so we needed to create the directory manually with the right permissions.
This means: Today's update to 2.5.0 **led to data loss** because the data directory was altered by the new rpm and the running daemon (2.4.0) suddenly lost all rights on its data directory and no data could be written!

Also, when packaging, please include something like
```
%postun systemd
%systemd_postun_with_restart influxdb.service
```
so the daemon gets restarted on upgrades.

**Comment by jules-ch** [comment]
I think I'm experiencing this issue.
After upgrading InfluxDB to 2.5.0.
This led to systemctl service to fail

I'm using a custom location for config.toml & data directories, permissions were not altered there. But I think permissions on binaries is making my service fails

**Comment by jules-ch** [comment]
Thanks @philfry, you saved me hours of debugging :+1:

**Comment by CendioMartin** [comment]
Bump. Same problem here after upgrade to 2.5.0

**Comment by nerijus** [comment]
I am including commands which helped me to start the service:
```
chmod 0755 /usr/bin/influxd /usr/lib/influxdb/scripts/influxd-systemd-start.sh /usr/lib/influxdb/scripts/init.sh /usr/share/influxdb/influxdb2-upgrade.sh 
chown influxdb:influxdb /var/lib/influxdb/
systemctl restart influxdb.service
```

**Comment by jules-ch** [comment]
The culprit seems to be #23683

**Comment by jeffreyssmith2nd** [comment]
2.5.1 has been released with updated RPMs and Debs. Upgrades from 2.4 -> 2.5.1 should see no issues, if you were on the short-lived 2.5 release, you may still need to run the commands in https://github.com/influxdata/influxdb/issues/23850#issuecomment-1300387095 to get everything up and running.

**Comment by jdstrand** [comment]
@jeffreyssmith2nd - should this issue be closed now that 2.5.1 is out?

**Comment by jeffreyssmith2nd** [comment]
Hey @jdstrand I was planning to leave this open for a few more days in case anyone somehow got on 2.5 to make it easier to find the instructions. Will close it out soon though.
